x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -271,6 +271,18 @@ const getSavedObjectTypes = (
         install_status: { type: 'keyword' },
         install_source: { type: 'keyword' },
         install_format_schema_version: { type: 'version' },
+        experimental_data_stream_features: {
+          type: 'nested',
+          properties: {
+            data_stream: { type: 'keyword' },
+            features: {
+              type: 'nested',
+              properties: {
+                synthetic_source: { type: 'boolean' },
```

Comment:
I felt it was better to be explicit about what features exist here, rather than just accepting any set of key/values. We'll need to expand this with additional settings moving forward.